### DISCUSSION TITLE
Language fallback in i18n routing

### DISCUSSION THREAD

**Discussion by motss** [amannn/next-intl#1222] [Q&A]
Hello everyone. I'm new to the `next-intl` space and still exploring this i18n library.

I have a use case where I want to be able to support 2-letter code prefix routing alongside other locales. I wonder if it is currently possible in `next-intl`.

Use case:
1. `/en` redirects to `/en-US`
1. `/ja` redirects to `/ja-JP`
1. `/ko` redirects to `/ko-KR`
1. `/en-US`, `/ja-JP`, `/ko-KR` renders the page without redirection.

```ts
const handleI18nRouting = createMiddleware({
  defaultLocale: 'en-US',
  localePrefix: {
    mode: 'always',
    prefixes: { 'en-US': '/en' } // Tried this but it does not seem to work for my use case
    // prefixes: { 'en-US': ['/en-US', '/en'] } // Expecting this but it does not accept arrays
  },
  locales: ['en-US', 'ja-JP', 'ko-KR'],
});
```

**Comment by amannn** [comment]
Sure, you can [compose the middleware](https://next-intl-docs.vercel.app/docs/routing/middleware#composing-other-middlewares) and add additional redirects as necessary on top!

In this case, you don't want to specify custom prefixes.

**Reply by motss** [reply]
> add additional redirects as necessary on top!
I thought of composing the middlewares which incurs an additional redirect.

However, the additional redirects can be eliminated if `next-intl` middleware does that internally. `next-intl` shares some similarities with the Astro i18n implementation and they provide a [fallback](https://docs.astro.build/en/guides/internationalization/#fallback) option that support this exact or similar use case.
